modules/xfem/src/efa/EFAElement2D.C

```diff
@@ -900,10 +954,74 @@ EFAElement2D::createChild(const std::set<EFAElement*> &CrackTipElements,
       childElem->setParent(this);
       _children.push_back(childElem);
 
+      std::vector<Point> local_embedded_node_coor;
+
+      for (unsigned int i = 0; i < this->getFragment(ichild)->numEdges(); ++i)
+      {
+        if (this->getFragment(ichild)->isEdgeInterior(i))
+        {
+          std::vector<EFANode*> master_nodes;
+          std::vector<double> master_weights;
+
+          for (unsigned int j = 0; j < 2; ++j)
+          {
+            this->getMasterInfo(this->getFragmentEdge(ichild,i)->getNode(j), master_nodes, master_weights);
+            Point coor(0.0, 0.0, 0.0);
+            for (unsigned int k = 0; k < master_nodes.size(); ++k)
+            {
+              EFANode * local = this->createLocalNodeFromGlobalNode(master_nodes[k]);
+              if (_num_edges == 3)
+                coor += master_weights[k] * _local_node_coor_tri[local->id()];
+              else
+                coor += master_weights[k] * _local_node_coor_quad[local->id()];
+            }
+            local_embedded_node_coor.push_back(coor);
+          }
+        }
+      }
+
+      Point normal(0.0, 0.0, 0.0);
+      Point origin(0.0, 0.0, 0.0);
+      Point normal2(0.0, 0.0, 0.0);
+      Point origin2(0.0, 0.0, 0.0);
+
+      if (local_embedded_node_coor.size())
+      {
+        Point cut_line = local_embedded_node_coor[1] - local_embedded_node_coor[0];
+        Real len = std::sqrt(cut_line.norm_sq());
+        if (len < 1.0e-10)
+          len = 1.0;
+        cut_line /= len;
+        normal = Point(cut_line(1), -cut_line(0), 0.0);
+        origin = (local_embedded_node_coor[0] + local_embedded_node_coor[1]) * 0.5;
+      }
+
+      if (local_embedded_node_coor.size() == 4)
+      {
+        Point cut_line = local_embedded_node_coor[3] - local_embedded_node_coor[2];
+        Real len = std::sqrt(cut_line.norm_sq());
+        if (len < 1.0e-10)
+          len = 1.0;
+        cut_line /= len;
+        normal2 = Point(cut_line(1), -cut_line(0), 0.0);
+        origin2 = (local_embedded_node_coor[2] + local_embedded_node_coor[3]) * 0.5;
+      }
+
       // get child element's nodes
       for (unsigned int j = 0; j < _num_nodes; ++j)
       {
-        if (_fragments[ichild]->containsNode(_nodes[j]))
+        Point p(0.0, 0.0, 0.0);
+        if (_num_edges == 3)
+          p = _local_node_coor_tri[j];
+        else
+          p = _local_node_coor_quad[j];
+
+        Point origin_to_point = p - origin;
+        Point origin2_to_point = p - origin2;
+
+        if (_fragments.size() == 1 && !shouldDuplicateForCrackTip(CrackTipElements))
+          childElem->setNode(j, _nodes[j]); // inherit parent's node
+        else if (origin_to_point * normal < 1.0e-10 && origin2_to_point * normal2 < 1.0e-10)
```

Comment:
This probably needs a comment to explain what's going on here.  Also, we should replace all of these usages of 1e-10 with a set tolerance variable.  It seems like we might already have one somewhere.